### ISSUE TITLE
[Feature Request] Run custom script stop() function on shutdown

### ISSUE THREAD

**Issue by dennisrijsdijk** [crowbartools/Firebot#2431]
**Describe the solution you'd like**
Currently custom scripts' stop() function doesn't get ran when Firebot shuts down. I would like script stop functions to be ran before shutdown so I can cleanup connections before shutdown completes.

**Additional context**
Currently in one of my scripts that may receive data up to several times per second, it'll throw several log errors prior to shutting down due to not being able to cleanup the connection.
```
[2024-02-28 16:10:25.0507] - error: Unhandled promise rejection 
{
  message: 'Object has been destroyed',
  stack: 'TypeError: Object has been destroyed\n' +
    '    at FrontendCommunicator.send (/home/dennisr/Programming/Firebot/build/backend/common/frontend-communicator.js:27:37)\n' +
    '    at handleTriggeredEvent (/home/dennisr/Programming/Firebot/build/backend/events/activity-feed-manager.js:29:26)\n' +
    '    at EventManager.<anonymous> (/home/dennisr/Programming/Firebot/build/backend/events/activity-feed-manager.js:49:5)\n' +
    '    at EventManager.emit (node:events:517:28)\n' +
    '    at EventManager.triggerEvent (/home/dennisr/Programming/Firebot/build/backend/events/EventManager.js:67:22)\n' +
    '    at triggerHeartRate (/home/dennisr/.config/Firebot/v5/profiles/bababa/scripts/hyperateScript.js:50:33)\n' +
    '    at WebSocket.message (/home/dennisr/.config/Firebot/v5/profiles/bababa/scripts/hyperateScript.js:133:51)\n' +
    '    at WebSocket.emit (node:events:517:28)\n' +
    '    at Receiver.receiverOnMessage (/home/dennisr/.config/Firebot/v5/profiles/bababa/scripts/hyperateScript.js:4747:20)\n' +
    '    at Receiver.emit (node:events:517:28)'
}
```

**Comment by dennisrijsdijk** [comment]
this error still shows up on the (currently) [latest v5 commit](https://github.com/crowbartools/Firebot/commit/9f627fa664c7d8ad396bafd9d0a2afa806fae069)

**Comment by zunderscore** [comment]
Released in v5.62.0
